ferretdb_packaging/definedockertags/main_test.go

```diff
@@ -0,0 +1,483 @@
+				"GITHUB_REF_NAME":   "ferretdb",
+				"GITHUB_REF_TYPE":   "branch",
+				"GITHUB_REPOSITORY": "OtherOrg/OtherRepo",
+				"INPUT_PG_VERSION":  "16",
+			},
+			expected: &result{
+				developmentImages: []string{
+					"ghcr.io/otherorg/otherrepo-dev:pr-docker-tag",
+				},
+			},
+		},
+
+		"push/main": {
+			env: map[string]string{
+				"GITHUB_BASE_REF":   "",
+				"GITHUB_EVENT_NAME": "push",
+				"GITHUB_HEAD_REF":   "",
+				"GITHUB_REF_NAME":   "main",
+				"GITHUB_REF_TYPE":   "branch",
+				"GITHUB_REPOSITORY": "FerretDB/documentdb",
+				"INPUT_PG_VERSION":  "16",
+			},
+		},
+		"push/main-other": {
+			env: map[string]string{
+				"GITHUB_BASE_REF":   "",
+				"GITHUB_EVENT_NAME": "push",
+				"GITHUB_HEAD_REF":   "",
+				"GITHUB_REF_NAME":   "main",
+				"GITHUB_REF_TYPE":   "branch",
+				"GITHUB_REPOSITORY": "FerretDB/documentdb",
+				"INPUT_PG_VERSION":  "16",
+			},
+		},
+
+		"push/tag/release": {
+			env: map[string]string{
+				"GITHUB_BASE_REF":   "",
+				"GITHUB_EVENT_NAME": "push",
+				"GITHUB_HEAD_REF":   "",
+				"GITHUB_REF_NAME":   "v0.102.0-ferretdb",
+				"GITHUB_REF_TYPE":   "tag",
+				"GITHUB_REPOSITORY": "FerretDB/documentdb",
+				"INPUT_PG_VERSION":  "16",
+			},
+			expected: &result{
+				developmentImages: []string{
+					"ferretdb/documentdb-dev:16-0.102.0-ferretdb",
+					"ferretdb/documentdb-dev:latest",
+					"ghcr.io/ferretdb/documentdb-dev:16-0.102.0-ferretdb",
+					"ghcr.io/ferretdb/documentdb-dev:latest",
+					"quay.io/ferretdb/documentdb-dev:16-0.102.0-ferretdb",
+					"quay.io/ferretdb/documentdb-dev:latest",
+				},
+				productionImages: []string{
+					"ferretdb/documentdb:16-0.102.0-ferretdb",
+					"ferretdb/documentdb:latest",
+					"ghcr.io/ferretdb/documentdb:16-0.102.0-ferretdb",
+					"ghcr.io/ferretdb/documentdb:latest",
+					"quay.io/ferretdb/documentdb:16-0.102.0-ferretdb",
+					"quay.io/ferretdb/documentdb:latest",
+				},
+			},
+		},
+		"push/tag/release-other": {
+			env: map[string]string{
+				"GITHUB_BASE_REF":   "",
+				"GITHUB_EVENT_NAME": "push",
+				"GITHUB_HEAD_REF":   "",
+				"GITHUB_REF_NAME":   "v0.102.0-ferretdb",
+				"GITHUB_REF_TYPE":   "tag",
+				"GITHUB_REPOSITORY": "OtherOrg/OtherRepo",
+				"INPUT_PG_VERSION":  "16",
+			},
+			expected: &result{
+				developmentImages: []string{
+					"ghcr.io/otherorg/otherrepo-dev:16-0.102.0-ferretdb",
+					"ghcr.io/otherorg/otherrepo-dev:latest",
+				},
+				productionImages: []string{
+					"ghcr.io/otherorg/otherrepo:16-0.102.0-ferretdb",
+					"ghcr.io/otherorg/otherrepo:latest",
+				},
+			},
+		},
+
+		"push/tag/release-rc-major-minor": {
+			env: map[string]string{
+				"GITHUB_BASE_REF":   "",
+				"GITHUB_EVENT_NAME": "push",
+				"GITHUB_HEAD_REF":   "",
+				"GITHUB_REF_NAME":   "v0.102.0-ferretdb-2.0.0-rc2",
+				"GITHUB_REF_TYPE":   "tag",
+				"GITHUB_REPOSITORY": "FerretDB/documentdb",
+				"INPUT_PG_VERSION":  "16.7", // set major and minor version
+			},
+			expected: &result{
```

Comment:
See this output for tagging `v0.102.0-ferretdb-2.0.0-rc2` with say postgresql version of `16.7`. It has `latest` tag, should we? It also has tag for pg `16` and pg `16.7`